File: Backend/FPAMS/FPAMS.Domain/Entities/Department.cs
using FPAMS.Domain.Common;

namespace FPAMS.Domain.Entities;

public class Department : BaseEntity
{
    public string DepartmentCode { get; set; } = string.Empty;

    public string DepartmentName { get; set; } = string.Empty;

    public bool IsActive { get; set; } = true;

    public ICollection<User> Users { get; set; } = new List<User>();
}

File: Frontend/fpams-web/src/types/department.ts
export interface Department {
    id: string;
    departmentCode: string;
    departmentName: string;
    isActive: boolean;
}

export interface CreateDepartmentRequest {
    departmentCode: string;
    departmentName: string;
    isActive: boolean;
}

export interface UpdateDepartmentRequest {
    id: string;
    departmentCode: string;
    departmentName: string;
    isActive: boolean;
}